boto3/s3/inject.py

```diff
@@ -56,3 +65,39 @@ def download_file(self, Bucket, Key, Filename, ExtraArgs=None,
     return transfer.download_file(
         bucket=Bucket, key=Key, filename=Filename,
         extra_args=ExtraArgs, callback=Callback)
+
+
+def bucket_upload_file(self, Filename, Key,
+                       ExtraArgs=None, Callback=None, Config=None):
+    """Upload a file to an S3 object."""
+    transfer = S3Transfer(self.meta.client, Config)
+    return transfer.upload_file(
+        filename=Filename, bucket=self.name, key=Key,
+        extra_args=ExtraArgs, callback=Callback)
+
+
+def bucket_download_file(self, Key, Filename,
+                         ExtraArgs=None, Callback=None, Config=None):
+    """Download an S3 object to a file."""
+    transfer = S3Transfer(self.meta.client, Config)
```

Comment:
Is there a reason we can't just use the `.meta.client` methods for upload_file/download_file?  Also seems like it would cut down on the duplicate boiler plate of creating an S3Transfer object for each of the four functions you've added.